extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkInputStreamGenerator.java

```diff
@@ -155,30 +300,33 @@ public void visitBuffers(final BufferListener listener) {
         @Override
         protected int getRawSize() {
             if (cachedSize == -1) {
-                cachedSize = 0;
+                MutableLong totalCachedSize = new MutableLong(0L);
                 if (sendValidityBuffer()) {
-                    cachedSize += getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME));
+                    totalCachedSize.add(getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME)));
                 }
 
                 // there are n+1 offsets; it is not assumed first offset is zero
-                if (!subset.isEmpty() && subset.size() == myOffsets.size() - 1) {
-                    cachedSize += myOffsets.size() * Integer.BYTES;
-                    cachedSize += myOffsets.get(subset.intSize(DEBUG_NAME)) - myOffsets.get(0);
+                if (!subset.isEmpty() && subset.size() == myByteStorage.offsets.size() - 1) {
+                    totalCachedSize.add(myByteStorage.offsets.size() * Integer.BYTES);
+                    for (int i = 0; i < myByteStorage.size(); i++) {
+                        totalCachedSize.add(myByteStorage.getByteArraySize(i));
+                    }
                 } else  {
-                    cachedSize += subset.isEmpty() ? 0 : Integer.BYTES; // account for the n+1 offset
+                    totalCachedSize.add(subset.isEmpty() ? 0 : Integer.BYTES); // account for the n+1 offset
                     subset.forAllRowKeyRanges((s, e) -> {
                         // account for offsets
-                        cachedSize += (e - s + 1) * Integer.BYTES;
+                        totalCachedSize.add((e - s + 1) * Integer.BYTES);
+
                         // account for payload
-                        cachedSize += myOffsets.get(LongSizedDataStructure.intSize(DEBUG_NAME, e + 1));
-                        cachedSize -= myOffsets.get(LongSizedDataStructure.intSize(DEBUG_NAME, s));
+                        totalCachedSize.add(myByteStorage.getPayloadSize((int)s, (int)e));
                     });
                 }
 
                 if (!subset.isEmpty() && (subset.size() & 0x1) == 0) {
                     // then we must also align offset array
-                    cachedSize += Integer.BYTES;
+                    totalCachedSize.add(Integer.BYTES);
                 }
+                cachedSize = LongSizedDataStructure.intSize("VarBinaryChunkInputStreamGenerator.getRawSize", totalCachedSize.longValue());
```

Comment:
Technically this int limit is on the payload; not including the cost for offsets. The offset buffer is limited to 2^32 offsets, not 2^32 bytes worth of offsets - so we should verify that the subset can be intSized(). We should sum up the payload separately.